glTF-Toolkit/src/GLTFLODUtils.cpp

```diff
@@ -274,15 +284,18 @@ namespace
                 material.name += nodeLodLabel;
                 AddIndexOffset(material.id, materialOffset);
 
-                AddIndexOffset(material.normalTexture.id, texturesOffset);
-                AddIndexOffset(material.occlusionTexture.id, texturesOffset);
-                AddIndexOffset(material.emissiveTextureId, texturesOffset);
+                AddIndexOffset(material.normalTexture.textureId, texturesOffset);
+                AddIndexOffset(material.occlusionTexture.textureId, texturesOffset);
+                AddIndexOffset(material.emissiveTexture.textureId, texturesOffset);
 
-                AddIndexOffset(material.metallicRoughness.baseColorTextureId, texturesOffset);
-                AddIndexOffset(material.metallicRoughness.metallicRoughnessTextureId, texturesOffset);
+                AddIndexOffset(material.metallicRoughness.baseColorTexture.textureId, texturesOffset);
+                AddIndexOffset(material.metallicRoughness.metallicRoughnessTexture.textureId, texturesOffset);
 
-                AddIndexOffset(material.specularGlossiness.diffuseTextureId, texturesOffset);
-                AddIndexOffset(material.specularGlossiness.specularGlossinessTextureId, texturesOffset);
+                if (material.HasExtension<KHR::Materials::PBRSpecularGlossiness>())
+                {
+                    AddIndexOffset(material.GetExtension<KHR::Materials::PBRSpecularGlossiness>().diffuseTexture.textureId, texturesOffset);

```

Comment:
Have you considered storing the result of GetExtension in a local variable rather than calling the function repeatedly?